internal/topology/variables/clusterclass_variable_validation_test.go

```diff
@@ -631,6 +632,76 @@ func Test_ValidateClusterClassVariable(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "fail if field is required below properties and sets a default",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						Properties: map[string]clusterv1.JSONSchemaProps{
+							"spec": {
+								Type:     "object",
+								Required: []string{"replicas"},
+								Example: &apiextensionsv1.JSON{
+									// replicas missing results in failure
+									Raw: []byte(`{"value": 100}`),
+								},
```

Comment:
Technically this case only ensure either Example or Default is validated cleanly. Should we split this into two cases?

Reply:
Examples not really required for these test cases so I've removed them.